azure/servicemanagement/__init__.py

```diff
@@ -1873,6 +1876,59 @@ def parse_response_for_async_op(response):
     return result
 
 
+def get_certificate_from_publish_settings(publish_settings_path, path_to_write_certificate, subscription_name=None):
+    '''
+    Writes a certificate file to the specified location.  This can then be used 
+    to instantiate ServiceManagementService.  Returns the subscription ID.
+
+    publish_settings_path: 
+        Path to subscription file downloaded from 
+        http://go.microsoft.com/fwlink/?LinkID=301775
+    path_to_write_certificate:
+        Path to write the certificate file.
+    subscription_name:
+        (optional)  Provide a subscription name here if you wish to use a 
+        specific subscription under the publish settings file.
+    '''
+    import base64
+    import OpenSSL.crypto as crypto
+
+    if publish_settings_path is None:
+        raise TypeError("publish_settings_path cannot be None")
+
+    if path_to_write_certificate is None:
+        raise TypeError("path_to_write_certificate cannot be None")
+
+    # parse the publishsettings file and find the ManagementCertificate Entry
+    tree = xml.etree.ElementTree.parse(publish_settings_path)
+    subscriptions = tree.getroot().findall("./PublishProfile/Subscription")
+    
+    if subscription_name is None:
+        # just take the first one in the file if they don't specify
+        subscription = subscriptions[0]
+    else:
+        for i in subscriptions:
+            if i.get('Name').lower() == subscription_name.lower():
+                subscription = i
+                break
```

Comment:
Also, are subscription names case-insensitive? At one point I actually had two subscriptions with identical names, so this may not even be the best way to identify them...